x-pack/plugins/ml/public/components/anomalies_table/anomaly_details.js

```diff
@@ -0,0 +1,326 @@
+  });
+
+  if (source.field_name !== undefined) {
+    items.push({
+      title: 'fieldName',
+      description: source.field_name
+    });
+  }
+
+  const functionDescription = source.function_description || '';
+  if (anomaly.actual !== undefined && showActualForFunction(functionDescription) === true) {
+    items.push({
+      title: 'actual',
+      description: formatValue(anomaly.actual, source.function)
+    });
+  }
+
+  if (anomaly.typical !== undefined && showTypicalForFunction(functionDescription) === true) {
+    items.push({
+      title: 'typical',
+      description: formatValue(anomaly.typical, source.function)
+    });
+  }
+
+  items.push({
+    title: 'job ID',
+    description: anomaly.jobId
+  });
+
+  items.push({
+    title: 'probability',
+    description: source.probability
+  });
+
+  // If there was only one cause, the actual, typical and by_field
+  // will already have been added for display.
+  if (causes.length > 1) {
+    causes.forEach((cause, index) => {
+      const title = index === 0 ? `${cause.entityName} values` : '';
+      let description = `${cause.entityValue} (actual ${formatValue(cause.actual, source.function)}, `;
+      description += `typical ${formatValue(cause.typical, source.function)}, probability ${cause.probability})`;
+      items.push({ title, description });
+    });
+  }
+
+  return items;
+}
+
+export class AnomalyDetails extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      showAllInfluencers: false
+    };
+  }
+
+  toggleAllInfluencers() {
+    this.setState({ showAllInfluencers: !this.state.showAllInfluencers });
+  }
+
+  renderDescription() {
+    const anomaly = this.props.anomaly;
+    const source = anomaly.source;
+
+    let anomalyDescription = `${getSeverity(anomaly.severity)} anomaly in ${anomaly.detector}`;
+    if (anomaly.entityName !== undefined) {
+      anomalyDescription += ` found for ${anomaly.entityName} ${anomaly.entityValue}`;
+    }
+
+    if ((source.partition_field_name !== undefined) &&
+        (source.partition_field_name !== anomaly.entityName)) {
+      anomalyDescription += ` detected in ${source.partition_field_name}`;
+      anomalyDescription += ` ${source.partition_field_value}`;
+    }
+
+    // Check for a correlatedByFieldValue in the source which will be present for multivariate analyses
+    // where the record is anomalous due to relationship with another 'by' field value.
+    let mvDescription = undefined;
+    if (source.correlated_by_field_value !== undefined) {
+      mvDescription = `multivariate correlations found in ${source.by_field_name}; `;
+      mvDescription += `${source.by_field_value} is considered anomalous given ${source.correlated_by_field_value}`;
+    }
+    return (
+      <React.Fragment>
+        <EuiText size="xs">
+          <h5>Description</h5>
+          {anomalyDescription}
+        </EuiText>
+        {mvDescription !== undefined &&
```

Comment:
nit, all conditions should really be wrapped in parenthesis, so this should be `(mvDescription !== undefined )`
There are quite a few examples like this so i'm going to stop flagging them up.